src/libraries/Common/src/Interop/Windows/Kernel32/Interop.FileScatterGather.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [StructLayout(LayoutKind.Explicit, Size = 8)]
+        internal unsafe struct FILE_SEGMENT_ELEMENT
+        {
+            [FieldOffset(0)]
+            public IntPtr Buffer;
```

Comment:
It's not correct for 32-bit, but I decided to not use these APIs for 32 bit because of the following limitation

From https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfilescatter#remarks:

> This function is not supported for 32-bit applications by WOW64 on Itanium-based systems.

The corresponding code:

https://github.com/dotnet/runtime/blob/73f07511127e4a47d5f57e8964519787ee735398/src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs#L303-L305

@stephentoub is that OK? If not, how can I detect that current process running on Itanium?

Reply:
I don't think anyone or anything cares about Itanium. .NET since Core certainly doesn't run on it anymore.

Reply:
I've confirmed offline that .NET Core does not support Itanium. I am going to modify the current logic and allow for using the method on x86 and ARM